nemo/collections/asr/parts/perturb.py

```diff
@@ -57,7 +57,8 @@ def perturb(self, data):
         impulse_record = self._rng.sample(self._manifest.data, 1)[0]
         impulse = AudioSegment.from_file(impulse_record.audio_file, target_sr=data.sample_rate)
         # logging.debug("impulse: %s", impulse_record['audio_filepath'])
-        data._samples = signal.fftconvolve(data.samples, impulse.samples, "full")
+        impulse_norm = (impulse.samples-min(impulse.samples))/(max(impulse.samples)-min(impulse.samples))
```

Comment:
Why subtract min instead of mean?

Reply:
mean would center the data, which is not we need, we need every impulse to be positive(>=0) and max to 1

Reply:
why does it need to be positive?
Maybe a better question then is: why are we subtracting the min?

Reply:
I would like to know the same , @nithinraok

Reply:
Unfortunately, there is no definitive answer I could put by citing references. But in Kaldi and related data augmentation works its recommended to keep the impulse values to be positive and normalized. Experimentally if you would listen to the audio, you can easily make out the difference, as normalized impulse responses keep the speaker sound the same and the same volume. If we think in convolution terms it means we are only adding up the positive impulse values with signal values.